### PR TITLE
Expand WebAssembly docs

### PR DESCRIPTION
**Pull request by GJZwiers**
This updates section 2.6 of the Deno manual and moves it to a small new chapter on WebAssembly (WASM). Both the synchronous and asynchronous APIs are now discussed with code examples given. In addition, some Rust code was added that should compile down to the bytes given in one of the code samples.

Some more explanations and links have been given all around and a page "Resources" has been added with links to other WASM resources. It may be helpful to Deno developers that not only want to use WASM but write their own WASM modules as well.

Question: Do I have to update `toc.json` or is it generated?

Any feedback is much appreciated :)

#12

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
@GJZwiers thank you very much for expanding these chapters! I'm happy with this, but I'll let another pair of eyes review before merging.

You will need to update `toc.json` manually.

File: webassembly.md
Comment (lucacasonato) [inline webassembly.md:7]:
```suggestion
Designed to be used alongside JavaScript to speed up key application components,
[WebAssembly](https://webassembly.org/) can have much higher, and more consistent execution speed than
JavaScript, similar to C, C++, or Rust. Deno can execute WebAssembly modules with the
same interfaces that
[browsers provide](https://developer.mozilla.org/en-US/docs/WebAssembly).
```

File: webassembly.md
Comment (lucacasonato) [inline webassembly.md:12]:
```suggestion
2. [Using the asynchronous WebAssembly APIs](webassembly/using_wasm_async.md)
```

File: webassembly/using_wasm.md
Comment (lucacasonato) [inline webassembly/using_wasm.md:3]:
```suggestion
To run WebAssembly in Deno, all you need is a binary to run. WebAssembly is a binary data format. This means that `.wasm`  files is not directly human readable, and is not intended to be written by hand. Instead a compiler for a language like Rust, C++, or Go _emits_ `.wasm` files.

The following binary
```

File: webassembly/using_wasm.md
Comment (lucacasonato) [inline webassembly/using_wasm.md:21]:
```suggestion
const main = wasmInstance.exports.main as CallableFunction;
```

File: webassembly/using_wasm.md
Comment (lucacasonato) [inline webassembly/using_wasm.md:29]:
```suggestion
1. Fetching the binary (usually in the form of a `.wasm` file, though we are
   using a simple byte array for now)
```

File: webassembly/using_wasm.md
Comment (lucacasonato) [inline webassembly/using_wasm.md:30]:
```suggestion
2. Compiling the binary into a `WebAssembly.Module` object
```

File: webassembly/using_wasm.md
Comment (lucacasonato) [inline webassembly/using_wasm.md:34]:
```suggestion
language that compiles down to WebAssembly instead of hand writing instructions. A number
```

File: webassembly/using_wasm_async.md
Comment (lucacasonato) [inline webassembly/using_wasm_async.md:1]:
```suggestion
## Using the asynchronous WebAssembly APIs
```

File: webassembly/using_wasm_async.md
Comment (lucacasonato) [inline webassembly/using_wasm_async.md:31]: Can you rewrite this example to use `await` instead of `then()`?

File: webassembly/using_wasm_async.md
Comment (lucacasonato) [inline webassembly/using_wasm_async.md:35]:
```suggestion
back to the less efficient `compile` and `instantiate` methods. See
```

File: webassembly/using_wasm_async.md
Comment (lucacasonato) [reply webassembly/using_wasm_async.md:1]: Really the WebAssembly instantiation APIs are all async. These APIs are just _streaming_. Maybe rename to `Using the streaming WebAssembly APIs`

File: webassembly/resources.md
Comment (lucacasonato) [inline webassembly/resources.md:22]:
```suggestion
If you plan to do a lot of work with Web APIs in Rust+WebAssembly, you may find
```

**Review by lucacasonato — COMMENTED** [review]
Very nice. A few comments.

**Comment by GJZwiers** [comment]
Thank you for the comments and feedback @bartlomieju and @lucacasonato ! I've addressed the issues and added the wasm pages to the table of contents

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @GJZwiers

**Pull request merged**
